base/docs/basedocs.jl

```diff
@@ -645,6 +645,36 @@ to be set after construction. See `struct` and the manual for more information.
 kw"mutable struct"
 
 """
+The `where` keyword creates a type that is an iterated union of other types, over all
+values of some variable. For example `Vector{T} where T<:Real` includes all `Vector`s
+where the element type is some kind of `Real` number.
+
+The variable bound defaults to `Any` if it is omitted:
+
+    Vector{T} where T    # short for `where T<:Any`
+
+Variables can also have lower bounds:
+
+    Vector{T} where T>:Int
+    Vector{T} where Int<:T<:Real
+
+There is also a concise syntax for nested `where` expressions. For example, this:
+
+    Pair{T, S} where S<:Array{T} where T<:Number
+
+can be shortened to:
+
+    Pair{T, S} where {T<:Number, S<:Array{T}}
+
+This form is often found on method signatures.
```

Comment:
I was thinking "on" as in "attached to", but yes I think any of those would be ok.